pkg/controller/mpi_job_controller.go

```diff
@@ -199,6 +199,16 @@ var (
 			Value: "-o ConnectionAttempts=10",
 		},
 	}
+	mpichEnvVars = []corev1.EnvVar{
+		{
+			Name:  "HYDRA_HOST_FILE",
+			Value: fmt.Sprintf("%s/%s", configMountPath, hostfileName),
+		},
+		{
+			Name:  "HYDRA_LAUNCH_EXTRA_ARGS",
+			Value: "-o ConnectionAttempts=10",
```

Comment:
Why do we need to set this option? Wouldn't `build/base/mpich-entrypoint.sh` be enough?

Reply:
Or, we might be able to use only this option and remove `build/base/mpich-entrypoint.sh`.

wdyt?

Reply:
Keep in mind I'm not an exper in eithere mpi-operator or MPI(CH), but I think:
1. we need entrypoint.sh to prepare the hostfile, so it needs to stay.
2. IntelMPI is based on MPICH, so I've just copied HYDRA_LAUNCH_EXTRA_ARGS as equivalant to Intel's I_MPI_HYDRA_BOOTSTRAP_EXEC_EXTRA_ARGS few lines above.
3. I would have guessed that the `max_retry=10` controls number of SSH connection attempts, and only after SHH connection is established, there are 10 connection attempts on Hydra/MPI side. So I think those params have different meaning.

Reply:
Thanks for the great clarification! I agree with you. Let's keep using both of them.

Reply:
These arguments affect the TCP+TLS connection, but they don't affect the DNS resolution, IIRC. So we might need both this and the entrypoint. But if we can get rid of the entry point, that would be better. Worth experimenting.

Reply:
> we need entrypoint.sh to prepare the hostfile, so it needs to stay.

The entry point doesn't prepare the hostfile. It just checks that all hosts listed in the hostfile are resolvable.